File: Backend/app/src/main/java/com/example/demo/repository/UserPlaylistRepository.java
package com.example.demo.repository;

import com.example.demo.model.UserPlaylist;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface UserPlaylistRepository extends JpaRepository<UserPlaylist, Long> {
    List<UserPlaylist> findByUserId(Long userId);
}

File: Mobile/app/src/main/java/com/example/myapplication/data/repository/UserPlaylistRepository.kt
package com.example.myapplication.data.repository

import com.example.myapplication.data.api.RetrofitInstance
import com.example.myapplication.data.model.UserPlaylist

class UserPlaylistRepository {
    suspend fun getUserPlaylist() : List<UserPlaylist>{
        return RetrofitInstance.userPlaylistApi.getUserPlaylist()
    }
}